src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -295,18 +284,13 @@ private void AddBody(InputParameter? bodyParameter, InputParameter? contentTypeR
             _protocolBodyParameter = bodyParameter.IsRequired
                 ? KnownParameters.RequestContent
                 : KnownParameters.RequestContentNullable;
-            _orderedParameters.Add(new ParameterChain(BuildParameter(bodyParameter), _protocolBodyParameter, _protocolBodyParameter));
 
-            if (contentTypeRequestParameter != null)
+            var convenienceBodyParameter = Parameter.FromInputParameter(bodyParameter, _typeFactory);
```

Comment:
`BuildParameter` already support to build an convenience parameter. Why do we define another series of functions?

Reply:
Sorry, I don't fully understand the question. `BuildParameter` has been relying on implicit check rather than explicit condition, which is the only option if at compile time we don't know if we have to change parameter type or not. However, it turned out that at the moment when parameter is created we do know explicitly if we need to change parameter type. E.g. (pseudo-code):
```cs
if (some_conditions) {
    Parameter.FromInputParameter(inputParameter, CreateTypeForParameter(...), _typeFactory)
} else {
    Parameter.FromInputParameter(inputParameter, _typeFactory)
}
```
instead of
```
var parameterType = some_conditions ? CreateTypeForParameter(...) : null;
BuildParameter(inputParameter, parameterType);
```